internal/broker/broker_test.go

```diff
@@ -827,6 +831,147 @@ func TestConcurrentIsAuthenticated(t *testing.T) {
 	}
 }
 
+func TestIsAuthenticatedAllowedUsersConfig(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		allowedUsers          map[string]bool
```

Comment:
Since this is just a set, it's better not to use booleans here or it may imply something else:

```suggestion
		allowedUsers          map[string]struct{}
```